### DISCUSSION TITLE
Intercom not found in Home Assistant

### DISCUSSION THREAD

**Discussion by H0d0rKF** [tsightler/ring-mqtt#585] [Support Q&A]
I installed the Add-On in Home Assistant but the Logs are saying that no device is Found. I have the Ring Intercom.

**Comment by tsightler** [comment]
Intercom is supported only in dev branch right now and "enable_cameras" configuration option must be set to true even if you have no cameras.

**Reply by H0d0rKF** [reply]
Ah, OK. When will it be in the main branch?

**Reply by tsightler** [reply]
I had hoped to get 5.2.0 out this past weekend, but was still tracking down various bugs, including one that was causing the code to crash.  You can always test the dev branch by setting the branch config option to "dev".

**Reply by H0d0rKF** [reply]
OK, thanks for your help! :)

**Comment by petoblat** [comment]
I have installed the addon setting the branch config option to dev but no entities appear.
in the addon log i see all intercom parameter but no in the mqtt integration in HA. 
mqtt mosquito broker is already configured and work well with zigbee2mqtt...

this an extract of the log:
"
2023-10-12T15:45:05.545Z ring-attr [Ingresso] ring/d53001d6-3d05-4173-846c-65743b2d476f/intercom/343ea4b155d8/wireless/attributes {"wirelessNetwork":"Wind3 HUB - Alex","wirelessSignal":-43}
2023-10-12T15:50:05.548Z ring-attr [Ingresso] ring/d53001d6-3d05-4173-846c-65743b2d476f/intercom/343ea4b155d8/info/state {"batteryLevel":95,"firmwareStatus":"Up to Date","lastUpdate":"2023-10-12T14:35:21Z","wirelessNetwork":"Wind3 HUB - Alex","wirelessSignal":-43}
2023-10-12T15:50:05.550Z ring-attr [Ingresso] ring/d53001d6-3d05-4173-846c-65743b2d476f/intercom/343ea4b155d8/battery/attributes {"batteryLevel":95}
2023-10-12T15:50:05.551Z ring-attr [Ingresso] ring/d53001d6-3d05-4173-846c-65743b2d476f/intercom/343ea4b155d8/wireless/attributes {"wirelessNetwork":"Wind3 HUB - Alex","wirelessSignal":-43}
"

can you help me please?
Thanks in advance

**Reply by tsightler** [reply]
Unfortunately, probably not.  I can only support ring-mqtt, and the small snippet of logs you posted would imply that everything there is working fine.  If the device/entities are not appearing in HA then it has to be something in the mosquitto or HA MQTT integration and there is already a tremendous amount of troubleshooting information out there for those.  In general, if everything is defaults, which is my recommendation if you don't otherwise know much about HA or MQTT, then it should work, if anything is non-default, i.e. you are using external MQTT, running stuff in containers and setting MQTT stuff manually, then that's for someone else to help with.

I'll share a little bit of experience though, one common mistake I see is users trying to leverage "guides" to setup things, but these "guides' usually assume that the thing you are trying to setup is the only thing that will exist.  For example, if you setup zigbee2mqtt, it will walk through the MQTT setup, but then if you install ring-mqtt and follow a "guide" it will have you install a different MQTT.  For example, I recently had a that was running moquitto in a container for zigbee2mqtt, but then install ring-mqtt addon in HA with defaults, of course this won't work because that is two different MQTT brokers and HA will only use a single MQTT broker.  You have to point both zigbee2mqtt and ring-mqtt to the same broker.

**Reply by roberrML** [reply]
Hi! I have the same problem, I see in addon's log that a intecom device is detected, but It doesn't appear as device.
I haven't configured anything weird, I just logged in addon interface. I'm using zigbee2mqtt and mosquito mqtt with default configuration (I think so) 
What do I have to share with you to be able to solve this? (Im not using docker)

Thanks in advance

**Reply by tsightler** [reply]
You can open an issue and send the full logs via email, but, realistically, if the addon is starting, connecting to the MQTT broker, sending the MQTT discovery and state messages, then there's not much I can do as the problem has to be on the other side of the MQTT broker.

**Reply by roberrML** [reply]
But, what can be missconfigured? or what can be happening, see logs from addon:
`2023-11-21T22:10:14.817Z ring-attr [Living room] ring/cf462fe1-bedf-4d27-8556-dc34d88e6b20/intercom/343ea4b088f4/wireless/attributes {"wirelessNetwork":"AFG_24","wirelessSignal":-60}`

**Reply by tsightler** [reply]
I don't know, there are all kinds of things it can be but, unfortunately, I just don't have time to help people troubleshoot MQTT and HA, if ring-mqtt is working, and sending messages to MQTT, that's where I stop.  Your single line of logging that you posted would appear to show that ring-mqtt is working, but certainly there should be far more logging than that.  Similar to the post above, there's zero chance of solving the problem with a few lines of logs, only full logs would have any hope.

There's TONS of information about troubleshooting MQTT and HA already out there, there's nothing special about troubleshooting ring-mqtt vs any other MQTT service, i.e. connect to the broker with some external tool like MQTT Explorer, see if the messages are coming thorough, if not, look into why, if so, check various things in HA to see why.

**Reply by roberrML** [reply]
I'm afraid I don't understand it. I mean, I don't understand how mqtt interacts with mqtt, or even why that's relevant. 
I can share all logs if you tell me what do you need. And I also read about "ring-mqtt vs any other MQTT service"
But I dont understand the basics I guess.

Do I have to configure mqtt url?
mqtt://auto_username:auto_password@auto_hostname
I don't have any configuration in my configuration.yaml for mqtt

Anyway, thanks a lot for your answer

**Reply by tsightler** [reply]
In general you should not need to configure the MQTT URL if you are using the standard Mosquitto addon with no special configuration, everything should be auto-detected.  As I mentioned above, you are welcome to open a support issue and provide logs and we can try to work through it, but, in general, if ring-mqtt is connecting to the broker and sending messages, the problem is not with ring-mqtt, but rather with something on the "other side".

It's really pretty simple, ring-mqtt connects to an MQTT broker, for Homeassistant this is usually the Mosquitto addon, and sends messages to that broker, any other MQTT client can connect to that broker and subscribe to receive those messages.  In the typical case it's the Homeassistant MQTT integration that connects to the Mosquitto addon to receive such messages, including discovery messages.

So, ring-mqtt connects to Mosquitto broker, sends discovery and state messages via MQTT, Home Assistant connects to MQTT broker and receives those messages.  If ring-mqtt can't connect to the MQTT broker (i.e. if the MQTT URL is wrong or broker is unreachable or whatever) it will log this and will not send state messages.  The single log line you posted is a state message, so that implies that ring-mqtt has connected to a broker successfully, thus the problem has to be on the receiving side.

Overall, my suggestion is to just restart everything and make sure it's all working.  Don't just restart HA, as that doesn't restart the addons, rather, when you choose to restart HA, click the advanced options and choose reboot system instead.

**Reply by roberrML** [reply]
After restart it works! Thanks
